packages/forge2d/lib/src/dynamics/body_def.dart

```diff
@@ -1,58 +1,92 @@
 import '../../forge2d.dart';
 
-/// A body definition holds all the data needed to construct a rigid body. You can safely re-use body
-/// definitions. Shapes are added to a body after construction.
+/// Holds all the data needed to construct a [Body].
+///
+/// You can safely re-use body definitions.
+///
+/// [Shape]s are added through [Fixture]s to a [Body] after construction via
+/// [Body.createFixture].
 class BodyDef {
```

Comment:
Personally, I would make this immutable. @spydon is there a reason why this is mutable?

Reply:
It makes sense having it as mutable I think, it doesn't really give much in Dart for this one to be immutable.
Being mutable you can just change the definition slightly if you are creating new groups of bodies, and you could also pool it if you'd need extreme amounts of them. You can also define it with cascading notation and build it up in several different parts of the code if need be.

Reply:
Makes sense. I feel that Flutter developers usually prefer immutable APIs over mutable APIs. I think this can be discussed in an issue or another PR.

Leaving it as it is for now.